python/mlx/nn/layers/activations.py

```diff
@@ -162,6 +163,60 @@ def gelu_fast_approx(x):
     return x * mx.sigmoid(1.773 * x)
 
 
+def glu(x: mx.array) -> mx.array:
+    """Applies the gated linear unit function.
+
+    This function splits the last dimension of the input into two halves
+    (a and b) and applies :math:`a * \sigma(b)`.
+
+    .. math::
+        \\textrm{GLU}(x) = a * \sigma(b)
+    """
+    a, b = mx.split(x, indices_or_sections=2, axis=-1)
+    return a * mx.sigmoid(b)
+
+
+def gated_activation(x: mx.array, activation_function: Callable) -> mx.array:
```

Comment:
I don't think we need this generalization at the moment. It's nice, and we can consider adding it in the future if we add more split style activations.